src/explore-education-statistics-common/src/modules/table-tool/components/FiltersForm.tsx

```diff
@@ -115,20 +125,30 @@ const FiltersForm = (props: Props & InjectedWizardProps) => {
   const handleSubmit = async (values: FormValues) => {
     setPreviousValues(values);
     try {
+      setTableQueryError(undefined);
       await onSubmit(values);
-      toggleTableSizeError.off();
       goToNextStep();
     } catch (error) {
       if (
-        isServerValidationError(error, 'QUERY_EXCEEDS_MAX_ALLOWABLE_TABLE_SIZE')
+        isServerValidationError(
+          error,
+          'QUERY_EXCEEDS_MAX_ALLOWABLE_TABLE_SIZE',
+        ) ||
+        isServerValidationError(error, 'REQUEST_CANCELLED')
       ) {
-        if (onTableSizeError) {
-          onTableSizeError(
+        const errorCode = getServerValidationError(
```

Comment:
Could refactor `getServerValidationError` to use a type parameter i.e.

```ts
export function getServerValidationError<T extends string = string>(...): T
```

Note that we're using a default type of `string` so that we don't need to explicitly set the type in every scenario i.e. we're happy receiving a plain string.

We can then use it like this (avoiding the `as`):

```ts
const errorCode = getServerValidationError<TableQueryErrorCode>(error);

if (errorCode) { ... }
```

Note that we should be using an `if` to be safe and check that the error message/code actually exists.

Reply:
Yup nice, nice.  Figured out the type guards that were in place, so that was a nice learning experience.  Good improvement.